src/pages/register-a-beacon/check-beacon-details.tsx

```diff
@@ -15,85 +15,144 @@ import {
 import { Details } from "../../components/Details";
 import { IfYouNeedHelp } from "../../components/Mca";
 import { GetServerSideProps, GetServerSidePropsContext } from "next";
-import { BeaconCacheEntry } from "../../lib/form-cache";
+import { BeaconCacheEntry } from "../../lib/formCache";
 import { updateFormCache } from "../../lib/middleware";
 import { ErrorSummary } from "../../components/ErrorSummary";
+import { FieldValidator } from "../../lib/fieldValidator";
 
 interface CheckBeaconDetailsProps {
-  manufacturerError?: boolean;
-  modelError?: boolean;
-  hexIdError?: boolean;
+  manufacturer: string;
+  model: string;
+  hexId: string;
+  needsValidation: boolean;
 }
 
-interface BeaconManufacturerInputProps {
-  isError: boolean;
-}
-
-interface BeaconModelInputProps {
-  isError: boolean;
-}
-
-interface BeaconHexIdSelectProps {
+interface FormInputProps {
+  value: string;
   isError: boolean;
+  errorMessages: Array<string>;
 }
 
 interface ErrorMessageProps {
   id: string;
   message: string;
 }
 
+const manufacturerField = new FieldValidator("manufacturer");
+
+manufacturerField
+  .should()
+  .containANonEmptyString()
+  .withErrorMessage("Manufacturer should not be empty");
+
+const modelField = new FieldValidator("model");
+
+modelField
+  .should()
+  .containANonEmptyString()
+  .withErrorMessage("Model should not be empty");
+
+const hexIdField = new FieldValidator("hexId");
+
+hexIdField
+  .should()
+  .containANonEmptyString()
+  .withErrorMessage("HEX ID should not be empty");
+
+hexIdField
+  .should()
+  .beExactly15Characters()
+  .withErrorMessage("Hex ID should be 15 characters long");
+
```

Comment:
Added a second validator so we can talk about giving the user specific feedback for exactly what's wrong in a particular field.  E.g.

"We understand one of the difficulties Linda's team has to deal with is non-UK encoded beacons making their way through the application process, or digits being slightly wrong.  We can now provide the user with really accurate and specific feedback about exactly what about their input needs to be double checked."